spec.bs

```diff
@@ -2198,9 +2195,19 @@ of the following global objects:
     1. Let |realm| be the result of [=creating a new script runner realm=] given
       {{InterestGroupScoringScriptRunnerGlobalScope}}.
     1. Let |browserSignalsJS| be |browserSignals| [=converted to ECMAScript values=].
-    1. TODO: convert |trustedScoringSignals| to a JS value.
+    1. Let |convertedTrustedScoringSignals| be a new empty [=map].
+    1. [=map/For each=] |key| -> |value| of |trustedScoringSignals|:
+      1. Let |valueString| be the result of [=serializing an Infra value to a JSON string=] given
+        |value|.
+      1. If the previous steps did not [=exception/throw=] an exception:
+        1. [=map/Set=] |convertedTrustedScoringSignals||key| to |valueString|.
+    1. Let |auctionConfigJS| be the result of [=converted to ECMAScript values|converting=]
+      |auctionConfigIDL| to an ECMAScript value.
+    1. Let |trustedScoringSignalsJS| be the result of

```

Comment:
I will say I'm a little confused what this is trying to do, largely because there doesn't seem to be any type information about what types are passed into the user-provided `scoreAd()`. It would be good to have some documentation about what the shape of this user-supplied API should be.

In any case, is the intention here to basically create a new plain old JavaScript object that we're passing into script? This really sounds like a record https://webidl.spec.whatwg.org/#idl-record, and per the last paragraph in that section which says we can treat an ordered map like a record, maybe we can just convert the ordered map (i.e., record) into an ECMAScript value with https://webidl.spec.whatwg.org/#ref-for-dfn-convert-idl-to-ecmascript-value%E2%91%A2%E2%91%A1. Do you agree?

Reply:
> Yes, it just tries to pass the auction config (in any way that's acceptable) to scoreAd().

I'm not sure I follow. My comment is about `trustedScoringSignalsJS` which is an ordered map in this PR. Auction config seems fine, since you're just calling the WebIDL convert algorithm on that dictionary, right? I'm asking about using a record for trustedScoringSignals/JS though, which seem to be just an ordered map of strings to strings, right? So I'm wondering if we can just call that a record and use that.

Reply:
Oh so sorry that I thought your comment was for the auction config line above.

Yes, all you said are absolutely right. The map's values are serialized so they're strings, so the ordered map can be implicitly treated as a record, and use WebIDL convert algorithm. Updated to that.